bot.py

```diff
@@ -0,0 +1,192 @@
+        return None
+
+    score, best_move = mini(white_team, bot_team, depth)
+
+    print(f'piece - {best_move[0]}. moved to - {best_move[1]}\n'
+          f'current score after the move is {score}')
+
+    piece, move_square = best_move
+    chess_utils.try_to_move(piece, move_square, bot_team, white_team)
+    return piece
+
+
+def mini(white_team: Team, bot_team: Team, depth):
+    best_score = 1000
+    best_move = None
+
+    if chess_utils.is_tie(bot_team, white_team):
+        return 0
+
+    if depth == 0:
+        return chess_utils.get_score(white_team, bot_team), best_move
+
+    for piece in bot_team.pieces:
+        if piece.is_eaten:
+            continue
+        valid_moves = piece.get_valid_move_squares()
+        for move_square in valid_moves:
+            try:
+                # If Didn't move, code wouldn't crash, just move to next move.
+                score_after_move = futuire_move(piece, move_square, white_team, bot_team, depth, is_bot_futiure_turn=True)
+
+                if score_after_move < best_score:
+                    best_move = (piece, move_square)
+                    best_score = score_after_move
+
+            except chess_utils.DidntMove:
+                pass
+
+    return best_score, best_move
+
+
+def maxi(white_team: Team, bot_team: Team, depth):
+    best_move = None
+    best_score = -1000
+
+    if chess_utils.is_tie(bot_team, white_team):
+        return 0
+
+    if depth == 0:
+        return chess_utils.get_score(white_team, bot_team), best_move
+
+    for piece in white_team.pieces:
+        if piece.is_eaten:
+            continue
+
+        valid_moves = piece.get_valid_move_squares()
+        for move_square in valid_moves:
+            try:
+                # If Didn't move code wouldn't crash, just move to next move.
+                with chess_utils.SaveMove(piece, move_square):
+                    score_after_move = futuire_move(piece, move_square, white_team, bot_team, depth, is_bot_futiure_turn=False)
+
+                    if score_after_move > best_score:
+                        best_move = (piece, move_square)
+                        best_score = score_after_move
+
+            except chess_utils.DidntMove:
+                pass
+
+    return best_score, best_move
+
+
+def futuire_move(piece, move_square, white_team, bot_team, depth, is_bot_futiure_turn):
+    next_move = maxi if is_bot_futiure_turn else mini
+    team_got_turn = bot_team if is_bot_futiure_turn else white_team
+    team_doesnt_got_turn = white_team if team_got_turn is bot_team else bot_team
+
+    with chess_utils.SaveMove(piece, move_square):
+        try:
+            chess_utils.try_to_move(piece, move_square, team_got_turn, team_doesnt_got_turn)
+        except chess_utils.MoveError:
+            raise chess_utils.DidntMove
+
+        score_after_move, _ = next_move(white_team, bot_team, depth - 1)
+
+    return score_after_move
+
+
+# This two fuctions not in use.
```

Comment:
Good comment, and just so you know I didn't looked at their code as they are currently not in use.